src/Models/NonhydrostaticModels/nonhydrostatic_model.jl

```diff
@@ -161,12 +161,12 @@ function NonhydrostaticModel(; grid,
     # First, we extract boundary conditions that are embedded within any _user-specified_ field tuples:
     embedded_boundary_conditions = merge(extract_boundary_conditions(velocities),
                                          extract_boundary_conditions(tracers),
-                                         extract_boundary_conditions(pressures),
```

Comment:
I don't fully understand the implications of this change. We're not trying to support user-defined BCs for pressure? Is that it?

Reply:
Yeah, before this PR, this line allowed users to (try) to put boundary conditions on pressure through the `boundary_conditions` kwarg. But after thinking about it, this can't have any effect because the velocity field is not prognostic on the boundaries. So it's useless...

This PR discontinuous the useless feature.